llama_index/indices/managed/zilliz/retriever.py

```diff
@@ -0,0 +1,74 @@
+import logging
+from typing import List, Optional
+
+import requests
+
+from llama_index.callbacks.base import CallbackManager
+from llama_index.constants import DEFAULT_SIMILARITY_TOP_K
+from llama_index.core import BaseRetriever
+from llama_index.indices.managed.zilliz.base import ZillizCloudPipelineIndex
+from llama_index.indices.query.schema import QueryBundle
+from llama_index.schema import NodeWithScore, QueryBundle, TextNode
+from llama_index.vector_stores.types import MetadataFilters
+
+logger = logging.getLogger(__name__)
+
+
+class ZillizCloudPipelineRetriever(BaseRetriever):
+    """A retriever built on top of Zilliz Cloud Pipeline's index."""
+
+    def __init__(
+        self,
+        index: ZillizCloudPipelineIndex,
+        search_top_k: int = DEFAULT_SIMILARITY_TOP_K,
+        filters: Optional[MetadataFilters] = None,
+        offset: int = 0,
+        output_metadata: list = [],
+        callback_manager: Optional[CallbackManager] = None,
+    ) -> None:
+        self.search_top_k = search_top_k
+        if filters:
+            exprs = []
+            for fil in filters.filters:
+                expr = f"{fil.key} == '{fil.value}'"
+                exprs.append(expr)
+            self.filter = " && ".join(exprs)
```

Comment:
is this only supporting exact match

Reply:
I will submit another pr to support other operations.